macros/sql/union.sql

```diff
@@ -10,7 +10,6 @@
     {% endif -%}
 
     {%- set column_override = column_override if column_override is not none else {} -%}
-    {%- set source_column_name = source_column_name if source_column_name is not none else '_dbt_source_relation' -%}
```

Comment:
@jtcohen6 — can you just sense check this for me (related to line #1). I looked at it a number of times and can't figure out why it's like this.

I guess similarly, why do we have `column_override=none` instead of `column_override={}`

Reply:
I think this is because of the python convention to set default arguments to `none`, always, out of fear of [mutable def](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments)[ault arguments](https://florimond.dev/blog/articles/2018/08/python-mutable-defaults-are-the-source-of-all-evil/).

It's less of an issue with strings and dicts than with arrays, so I think it's _probably_ fine to make this change.